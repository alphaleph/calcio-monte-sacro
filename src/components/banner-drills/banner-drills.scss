@import '../../pages/globals.scss';

.banner-drills {
    background: linear-gradient(to top right, #48CA73 70%, #69D38C);
    border-top: 1px solid $holy-yellow-team;
    border-bottom: 1px solid $holy-yellow-team;

}
// 69D38C
// 58CE7F
// 42B869

.banner-drills-title {
    text-align: center;
    color: $black-main;
    font-size: 3rem;
    font-weight: bold;
    margin-bottom: 2rem;
}

.banner-drills-note {
    font-size: 3rem;
    padding-bottom: .5rem;
}

.banner-drills-button {
    border-color: $granite-grey-team !important;
    font-weight: bold;
    font-size: 1.25rem !important;
    transform: scale(1);
    transition: transform .05s ease-out;
}

.banner-drills-button:hover,
.banner-drills-button:focus {
    border-color: $granite-grey-team !important;
    text-decoration: underline $holy-yellow-team;
    // font-size: 1.5rem;
    transform: scale(1.02);
}