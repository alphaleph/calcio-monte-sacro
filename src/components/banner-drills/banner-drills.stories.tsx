import React from 'react';
import { storiesOf } from '@storybook/react';
import { BannerDrills } from './banner-drills';
import '../../pages/index.scss';

storiesOf('BannerDrills', module).add('default', () => <BannerDrills />);
