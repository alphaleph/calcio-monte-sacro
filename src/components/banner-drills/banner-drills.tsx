import React from 'react';
import { Link } from 'gatsby';
import { SITE_URL } from '../../constants';
import './banner-drills.scss';

export const BannerDrills = () => {
    return (
        <section className="banner-drills section">
            <div className="container">
                <h2 className="banner-drills-title">
                    The Techniques of the Trade...
                </h2>
                <div className="banner-drills-note columns is-mobile has-text-centered">
                    <p className="column is-one-third">ðŸ‘‰ðŸ¦µ</p>
                    <p className="column is-one-third">ðŸ‘‰âš½</p>
                    <p className="column is-one-third">ðŸ‘‰ðŸ¥…</p>
                </div>
                <div className="columns">
                    <div className="column is-half is-offset-one-quarter">
                        <Link to={`/drills`}>
                            <button
                                className="banner-drills-button button is-fullwidth has-background-team-granite has-text-team-holy"
                                type="button"
                            >
                                SEE DRILLS
                            </button>
                        </Link>
                    </div>
                </div>
            </div>
        </section>
    );
};
