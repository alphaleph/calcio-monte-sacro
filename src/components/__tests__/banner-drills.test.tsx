import React from 'react';
import { BannerDrills } from '../banner-drills/banner-drills';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

const title = 'The Techniques of the Trade...';
const buttonText = 'SEE DRILLS';

describe('Banner Drills', () => {
    it('renders without crashing', () => {
        const { getByText } = render(<BannerDrills />);
        expect(getByText(title)).toBeInTheDocument();
        expect(getByText(buttonText)).toBeInTheDocument();
    });

    it('renders correctly', () => {
        const { getByText } = render(<BannerDrills />);
        expect(getByText(title)).toBeVisible();
        expect(getByText(buttonText)).toBeVisible();
    });

    it('has not changed', () => {
        const { asFragment } = render(<BannerDrills />);
        expect(asFragment()).toMatchSnapshot();
    });
});
