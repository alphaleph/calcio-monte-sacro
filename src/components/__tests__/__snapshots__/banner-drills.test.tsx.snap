// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Drills has not changed 1`] = `
<DocumentFragment>
  <section
    class="banner-drills section"
  >
    <div
      class="container"
    >
      <h2
        class="banner-drills-title"
      >
        The Techniques of the Trade...
      </h2>
      <div
        class="banner-drills-note columns is-mobile has-text-centered"
      >
        <p
          class="column is-one-third"
        >
          ðŸ‘‰ðŸ¦µ
        </p>
        <p
          class="column is-one-third"
        >
          ðŸ‘‰âš½
        </p>
        <p
          class="column is-one-third"
        >
          ðŸ‘‰ðŸ¥…
        </p>
      </div>
      <div
        class="columns"
      >
        <div
          class="column is-half is-offset-one-quarter"
        >
          <a
            href="/drills"
          >
            <button
              class="banner-drills-button button is-fullwidth has-background-team-granite has-text-team-holy"
              type="button"
            >
              SEE DRILLS
            </button>
          </a>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
